src/main/java/org/apache/ibatis/builder/annotation/MapperAnnotationBuilder.java

```diff
@@ -488,11 +488,8 @@ private SqlSource getSqlSourceFromAnnotations(Method method, Class<?> parameterT
 
   private SqlSource buildSqlSourceFromStrings(String[] strings, Class<?> parameterTypeClass, LanguageDriver languageDriver) {
     final StringBuilder sql = new StringBuilder();
-    for (String fragment : strings) {
-      sql.append(fragment);
-      sql.append(" ");
-    }
-    return languageDriver.createSqlSource(configuration, sql.toString().trim(), parameterTypeClass);
+    return languageDriver.createSqlSource(configuration,
+      String.join(" ", strings).trim(), parameterTypeClass);
```

Comment:
@a827871781 I'm applying this change separately.  The sql builder needed removed as well.  This will be on master shortly.